### DISCUSSION TITLE
[handleAbortError] AI response error; aborting request: Prompt token count of 11955 exceeds max token count of 8187.

### DISCUSSION THREAD

**Discussion by ryan-kettrey** [danny-avila/LibreChat#3847] [Troubleshooting] [answered]
Really confused here. We are getting the error in the title when we use Mistral Instruct 7B v2 through Bedrock through Lite LLM proxy.  In the logs I see:

`LibreChat         | 2024-08-29 19:46:55 error: [handleAbortError] AI response error; aborting request: Prompt token count of 11955 exceeds max token count of 8187.`

However, if we go to the Bedrock playground directly, we don't get this error, it works great.

We didn't change the defaults in the Lite LLM proxy configuration (see below) or in the LibreChat configurations, so my assumption is it defaults to the max token settings defined here (https://github.com/BerriAI/litellm/blob/main/model_prices_and_context_window.json). 

Specifically, this block:

`"bedrock/us-east-1/mistral.mistral-7b-instruct-v0:2": {`
`        "max_tokens": 8191,`
`        "max_input_tokens": 32000,`
`        "max_output_tokens": 8191,`
`        "input_cost_per_token": 0.00000015,`
`        "output_cost_per_token": 0.0000002,`
`        "litellm_provider": "bedrock",`
`        "mode": "chat"`
`    },`

From this you can see the max input tokens is 32k. In the bedrock playground, we are able to submit the 11955 tokens without issue. Our LiteLLM proxy config is:

`- model_name: mistral-7b-instruct-v2`
`    litellm_params:`
`      model: bedrock/mistral.mistral-7b-instruct-v0:2`
`      aws_region_name: us-east-1`
`      aws_access_key_id: ...`
`      aws_secret_access_key: ...`

I also tried it with `model: bedrock/us-east-1/mistral.mistral-7b-instruct-v0:2` for the heck of it and got the same results.

I'm sort of thinking I need to increase the `max_tokens` in this json file, but, not feeling 100% confident that is the fix. I did try adding a model preset definition of `max_tokens: 12000` in librechat.yaml to see if that overrides, but, I still get an error.

I did see this discussion [1642](https://github.com/danny-avila/LibreChat/discussions/1642), but seemed to indicate for that one it was missing the model. In my case the model is defined already.

**Comment by danny-avila** [comment, marked as answer]
@ryan-kettrey the max context windows for AWS models are not yet defined by the system. However, this will change soon, and eventually a way to configure your own defaults.

**Reply by ryan-kettrey** [reply]
So, interestingly, Claude 3 Haiku with AWS bedrock is handling this 11955 token sized prompt, using LIbreChat and Lite LLM Proxy.

I'm assuming this just means the AWS side has a higher default context window for that model.

**Reply by danny-avila** [reply]
@ryan-kettrey it's not an issue with AWS-side, LibreChat also defines limits server-side for known models, anthropic being one already well known. this needs an update I'm hoping to make this weekend.

**Reply by ryan-kettrey** [reply]
Ah, I see. Thanks.

**Reply by ryan-kettrey** [reply]
If it something you want to farm out and have a clear idea how it works, we may be able to get into it as well. I poked around here a bit [bedrock.py](https://github.com/BerriAI/litellm/blob/main/litellm/llms/bedrock.py) and thinking maybe that's the area you are thinking of.
